### DISCUSSION TITLE
Query on white spaces in identifiers

### DISCUSSION THREAD

**Discussion by pkarandikarklp** [ezylang/EvalEx#339] [General]
I know that usually languages do not allow white spaces in the identifiers like variable names, function names etc.
I would like to know if there is any way where I can have spaces in the expression variables 
e.g.
In the below expression
strcat(**first name**, **last name**) == "abc xyz" where the variables in the expression contain spaces

**Comment by uklimaschewski** [comment]
It is not possible to use variable names with white spaces directly in an expression, it violates the syntax.
But maybe indirectly. You should be able to pass a variable with white spaces to the expression.
If you then define a function to retrieve the value by name, this function could retrieve the value from the data store.

`Expression expression = new Expression("getValue(\\"first name\\") + \\" \\" + getValue(\\"last name\\"));
(You can concat string in an Expression with the + operator)

Definition of the custom getValue() function should be easy.
